### PR TITLE
Fix `ResultCollection._get_cache` with multielement tensors

### LINKED ISSUE #9528: Error with metrics with `None` averaging
## 🐛 Bug

There is an issue when using torch metrics with an averaging method of None. It seems that when the trainer is checking if the metrics are ready, as they are tensors of multiple elements (one for each class) they cannot be interpreted as booleans, hence the process crashes. 

### To Reproduce

https://colab.research.google.com/drive/1Hil03eDMHynRDNa0dG9SAVuPqmc0w2vo?usp=sharing

### Expected behavior

Just continue to the second epoch.

### Environment

```
* CUDA:
	- GPU:
		- Tesla K80
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.9.0+cu102
	- pytorch-lightning: 1.4.6
	- tqdm:              4.62.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.11
	- version:           #1 SMP Sat Jun 5 09:50:34 PDT 2021
```

### Additional context

None.

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes #9528

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -349,6 +349,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed `BasePredictionWriter` not returning the batch_indices in a non-distributed setting ([#9432](https://github.com/PyTorchLightning/pytorch-lightning/pull/9432))
 
 
+- Fixed check on torchmetrics logged whose `compute()` output is a multidimensional tensor ([#9582](https://github.com/PyTorchLightning/pytorch-lightning/pull/9582))
```
Comment (carmocca) [inline CHANGELOG.md:352]:
```suggestion
- Fixed check on torchmetrics logged whose `compute()` output is a multielement tensor ([#9582](https://github.com/PyTorchLightning/pytorch-lightning/pull/9582))
```

File: tests/core/test_metric_result_integration.py [hunk 2]
```diff
@@ -544,3 +549,16 @@ def on_train_epoch_end(self) -> None:
 
     trainer = Trainer(default_root_dir=tmpdir, max_epochs=2, limit_train_batches=2, limit_val_batches=0)
     trainer.fit(model)
+
+
+def test_metric_result_computed_check():
+    """Unittest ``_get_cache`` with multidimensional tensors."""
```
Comment (carmocca) [inline tests/core/test_metric_result_integration.py:555]:
```suggestion
    """Unittest ``_get_cache`` with multielement tensors."""
```

**Review by tchaton — APPROVED** [review]
Great catch !

**Review by awaelchli — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:
